project/Dependencies.scala

```diff
@@ -63,4 +63,17 @@ object Dependencies {
   lazy val exi = Seq(
     "com.siemens.ct.exi" % "exificient" % "1.0.7",
   )
+
+  lazy val xjc = 
```

Comment:
This `xjc` dependency list is assigned to both `xjcLibs` and `libraryDependencies`. This is fine for `xjcLibs` since that setting only defines dependencies needed to run the xjc source generator. But I don't think we can have dependencies added to `libraryDependencies` that are conditional on the java version, since the version of Java we build with might be different than the version of Java a used at runtime.

In fact, since releases are built and published with Java 8, these won't be listed as actual dependencies for our official releases. So people using Java 11 or later won't have them as a dependency.

But maybe that's okay and none of these should be added to `libraryDependencies` in the first place? Maybe we document (or maybe it's obvious to JAXB users?) that if a user wants to use these xjc generated files they need to depend on those xjc libraries? Or maybe we just always require them regardless of java version?

Is this the result of https://github.com/apache/daffodil/pull/1057#discussion_r1339338337

Note, another win for scalaxb--I don't think it has any dependency complications like this.

Reply:
Also, this adds new dependencies. Have you verified that the license are ASF compatible and if changes are needed to NOTICE/LICENSE?

Reply:
I believe we will need to make updates to the NOTICE/LICENSE files.

SbtXjcPlugin is BSD-3
jaxb-xjc and jaxb-impl are EDL 1.0
activation is CDDL 1.0 - this is the most restrictive, but I think we're okay because we're open source.

On the other hand, if we switch to scalaxb, that uses MIT.

However, there are dependencies listed on the github page that I don't see in Steve's code snippet. In case we need those:
org.dispatchhttp.dispatch-core - LGPL 3.0
javax.xml.bind.jaxb-api - CDDL 1.0
org.scala-lang.modules.scala-xml - Apache 2.0
org.scala-lang.scala-parser-combinators - Apache 2.0.

Reply:
The xjc dependencies are fine, but CDDL is category B we do need to mention CDDL so needs to be mentioned in a readme.

We already depend on scala-xml and scala-parser-combinators. LGPL is category X and is not allowed by ASF, but I don't think it's a mandatory dependency. Dispatch http looks like an http communication library library that scalaxb can generate code to support, but it looks optional to me. For example, I think it can be disabled with:

```scala
Compile / scalaxb / scalaxbGenerateDispatchClient := false
```

Similarly scalaxb also supports http4s and gigahorse, which I think are also optional with similar keys to disabled (disabled by default).  I'm also not sure where scalaxb actually uses the jaxb-api. I don't see it mentioned in the generated code. Maybe that's some an optional dependency as well?

But either way I think we're fine, we just need to make sure to update LICENSE/NOTICE appropriately if CDDL is included.